lib/rules/forbid-component-props.js

```diff
@@ -38,6 +45,9 @@ module.exports = {
           defaultAllowOnDOM: {
             type: 'boolean'
           },
+          defaultAllowOnComponents: {
+            type: 'boolean'
+          },
           forbid: {
```

Comment:
Wouldn't that mean the entire plugin would have to switch into whitelist mode?  Would that be useful to anyone?